### PR TITLE
feat: add query protection feature

### LINKED ISSUE #274: add query protection feature
### Is your feature request related to a problem? Please describe.（如果该特性或者功能和已知某个问题相关，请描述问题是什么）

Throws an error if the query has no time condition.

### Describe the solution you'd like（问题的解决办法是什么）

1. Configuration items need to be added, and this protection feature is turned off by default
2. Throws an error if the query has no time condition.

### Describe alternatives you've considered（描述您考虑过的替代方案）

_No response_

### Additional context（其他补充内容）

_No response_

### PR DESCRIPTION
**Pull request by big-dust**
### What problem does this PR solve?
close #274 

- When query protection is enabled, queries that do not have time limits will return errors.

### How Has This Been Tested?
1. When configuring the message time-filter-protection = true:

1.1 An error message will be returned if the time range is not limited.
```sql
select * from [measurement] 
```

1.2 A time limit can be smoothly queried.
 ```sql
select * from [measurement] where time < now()
```

2. When configuring the message time-filter-protection = false:
- The query is not affected by the time range or not.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by shilinlee — CHANGES_REQUESTED** [review]

File: open_src/influx/httpd/config/config.go [hunk 1]
```diff
@@ -37,71 +37,73 @@ const (
...
+	WriteRequestRateLimit         int            `toml:"write-request-ratelimit"`
+	EnqueuedQueryTimeout          toml.Duration  `toml:"enqueued-query-timeout"`
+	TLS                           *tls.Config    `toml:"-"`
+	WhiteList                     string         `toml:"white_list"`
+	SlowQueryTime                 toml.Duration  `toml:"slow-query-time"`
+	ParallelQueryInBatch          bool           `toml:"parallel-query-in-batch-enabled"`
+	QueryMemoryLimitEnabled       bool           `toml:"query-memory-limit-enabled"`
+	ChunkReaderParallel           int            `toml:"chunk-reader-parallel"`
+	ReadBlockSize                 toml.Size      `toml:"read-block-size"`
+	DisableQueryWithoutTimeFilter bool           `toml:"disable-query-without-time-filter"`
```
Comment (shilinlee) [inline open_src/influx/httpd/config/config.go:79]: Use global variables, see `querySeriesLimit ` or `LogRowsRuleSwitch `

https://github.com/openGemini/openGemini/blob/a18abae75d436e25df56d124176c8fb532ee37bb/lib/syscontrol/syscontrol.go#L93

https://github.com/openGemini/openGemini/blob/a18abae75d436e25df56d124176c8fb532ee37bb/lib/syscontrol/syscontrol.go#L115

**Review by shilinlee — CHANGES_REQUESTED** [review]

File: open_src/influx/httpd/config/config.go (hunk 1 above)
Comment (shilinlee) [reply open_src/influx/httpd/config/config.go:79]: change to:
```
DisableNoTimeFilter bool           `toml:"disable-query-without-time-filter"`
```

File: lib/syscontrol/syscontrol.go [hunk 2]
```diff
@@ -225,6 +235,13 @@ func ProcessRequest(req netstorage.SysCtrlRequest, resp *strings.Builder) (err e
 		executor.SetEnableForceBroadcastQuery(enabled)
 		res := "\n\tsuccess"
 		resp.WriteString(res)
+	case disableNoTimeFilter:
+		if switchon := req.Param()["switchon"]; switchon != "true" && switchon != "false" {
```
Comment (shilinlee) [inline lib/syscontrol/syscontrol.go:239]: Why not use `GetBoolValue`?

File: open_src/influx/httpd/config/config.go [hunk 3]
```diff
@@ -76,6 +76,7 @@ type Config struct {
 	QueryMemoryLimitEnabled bool           `toml:"query-memory-limit-enabled"`
 	ChunkReaderParallel     int            `toml:"chunk-reader-parallel"`
 	ReadBlockSize           toml.Size      `toml:"read-block-size"`
+	DisableNoTimeFilter     bool           `toml:"disable-query-without-time-filter"`
```
Comment (shilinlee) [inline open_src/influx/httpd/config/config.go:79]: add the definition `disable-query-without-time-filter` to the config file

**Review by shilinlee — CHANGES_REQUESTED** [review]

File: lib/syscontrol/syscontrol.go (hunk 2 above)
Comment (big-dust) [reply lib/syscontrol/syscontrol.go:239]: soryy , i will change it

File: open_src/influx/httpd/config/config.go (hunk 3 above)
Comment (big-dust) [reply open_src/influx/httpd/config/config.go:79]: > add the definition `disable-query-without-time-filter` to the config file
sorry, i will add it immediately

File: lib/syscontrol/syscontrol.go (hunk 2 above)
Comment (big-dust) [reply lib/syscontrol/syscontrol.go]: > Why not use `GetBoolValue`?

sorry , I will change it

File: config/openGemini.conf [hunk 4]
```diff
@@ -66,6 +66,7 @@
   # https-enabled = false
   # https-certificate = ""
   # https-private-key = ""
+  # disable-query-without-time-filter = false
```
Comment (scuzyj) [inline config/openGemini.conf:69]: Configuration item name: disable-query-without-time-filter need a better name, time-filter-protection?

File: app/ts-sql/sql/server.go [hunk 5]
```diff
@@ -162,6 +162,7 @@ func NewServer(conf config.Config, cmd *cobra.Command, logger *Logger.Logger) (a
 	s.initStatisticsPusher()
 	s.httpService.Handler.StatisticsPusher = s.statisticsPusher
 	syscontrol.SetQueryParallel(int64(c.HTTP.ChunkReaderParallel))
+	syscontrol.SetDisableNoTimeFilter(c.HTTP.DisableNoTimeFilter)
```
[the remark this answers is not in the document]
Comment (big-dust) [reply app/ts-sql/sql/server.go:165]: > 1、A better name needs to be defined for the SetDisableNoTimeFilter function, SetTimeFilterProtection? 2、Member Variable DisableNoTimeFilter of HTTP struct，also need a better name, TimeFilterProtection?

yes，I agree with you

**Review by shilinlee — APPROVED** [review]

File: lib/syscontrol/syscontrol.go [hunk 6]
```diff
@@ -71,20 +72,21 @@ curl -i -XPOST 'http://127.0.0.1:8086/debug/ctrl?mod=force_broadcast_query&enabl
...
+	ChunkReaderParallel  = "chunk_reader_parallel"
+	BinaryTreeMerge      = "binary_tree_merge"
+	PrintLogicalPlan     = "print_logical_plan"
+	SlidingWindowPushUp  = "sliding_window_push_up"
+	ForceBroadcastQuery  = "force_broadcast_query"
+	Failpoint            = "failpoint"
+	Readonly             = "readonly"
+	LogRows              = "log_rows"
+	verifyNode           = "verifynode"
+	TimeFilterProtection = "time-filter-protection"
```
Comment (scuzyj) [inline lib/syscontrol/syscontrol.go:89]: The naming style of the time-filter-protection variable is different from that of other variables. The hyphen (-) is changed to underscore (_).

File: lib/syscontrol/syscontrol.go [hunk 7]
```diff
@@ -106,6 +108,14 @@ func GetQuerySeriesLimit() int {
 	return querySeriesLimit
 }
 
+func SetTimeFilterProtection(swithon bool) {
```
Comment (scuzyj) [inline lib/syscontrol/syscontrol.go:111]: it is advised to name the "switch on" to "switch" or "enable" variable.

File: lib/syscontrol/syscontrol.go [hunk 8]
```diff
@@ -225,6 +235,14 @@ func ProcessRequest(req netstorage.SysCtrlRequest, resp *strings.Builder) (err e
 		executor.SetEnableForceBroadcastQuery(enabled)
 		res := "\n\tsuccess"
 		resp.WriteString(res)
+	case TimeFilterProtection:
+		switchon, err := GetBoolValue(req.Param(), "switchon")
```
Comment (shilinlee) [inline lib/syscontrol/syscontrol.go:239]: It's a good practice to add some test cases.

**Review by shilinlee — CHANGES_REQUESTED** [review]

File: lib/syscontrol/syscontrol.go (hunk 7 above)
Comment (big-dust) [reply lib/syscontrol/syscontrol.go]: > It is advised to name the "switchon" to "enable" variable.

yes

File: lib/syscontrol/syscontrol.go (hunk 6 above)
Comment (big-dust) [reply lib/syscontrol/syscontrol.go]: > The naming style of the time-filter-protection variable is different from that of other variables. The hyphen (-) is changed to underscore (_).
ok

File: lib/syscontrol/syscontrol.go (hunk 8 above)
Comment (big-dust) [reply lib/syscontrol/syscontrol.go]: > It's a good practice to add some test cases.

ok

File: lib/syscontrol/syscontrol.go [hunk 9]
```diff
@@ -71,20 +72,21 @@ curl -i -XPOST 'http://127.0.0.1:8086/debug/ctrl?mod=force_broadcast_query&enabl
 */
```
Comment (shilinlee) [inline lib/syscontrol/syscontrol.go:72]: Add the usage for `mod=time_filter_protection`.

**Review by shilinlee — CHANGES_REQUESTED** [review]

File: lib/syscontrol/syscontrol.go (hunk 9 above)
Comment (big-dust) [reply lib/syscontrol/syscontrol.go:72]: > Add the usage for `mod=time_filter_protection`.

ok

**Review by shilinlee — APPROVED** [review]

**Review by scuzyj — APPROVED** [review]

**Pull request merged**